Assets/Scripts/MusicManager.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using PixelCrushers.DialogueSystem;
+using System;
+
+/// <summary>
+/// Manages the background music at runtime
+/// </summary>
+public class MusicManager : MonoBehaviour
+{
+    /// <summary>
+    /// The current state of the town, based on quests completed
+    /// Determines which of the four tracks to play
+    /// </summary>
+    private enum TownMusicState { Cursed, Dreary, Light, Cheery };
+
+    /// <summary>
+    /// Which theme should currently be playing?
+    /// </summary>
+    public enum CurrentMusic { Town, Cave, Forest, Castle };
```

Comment:
I like the name `TownMusicState` better than `CurrentMusic` for an enum, because `CurrentMusic` doesn't sound like it's defining a type, it sounds like it's a variable name. Maybe `CurrentMusicType`?

Reply:
Maybe it's the "current" that's the problem in the name?

Reply:
`MusicType`? I don't know!